### DISCUSSION TITLE
Indication of missing/not configured layout slots in the property pane

### DISCUSSION THREAD

**Discussion by jblueml** [microsoft-search/pnp-modern-search#3672] [Ideas]
Hi all. 

I would find it a helpful feature to display an indication of missing/not configured layout slots in the property pane.

This means: a check would be done against the configured layout slots and the slots being referred within the currently selected layout or within results types. If there are slots being used in layout or result types, that have not been configured properly, a warning message will be displayed.

The indication would take place at the following 3 locations:
- below the layout slots "Customize" button on page 1 (common warning-message if there are missing slots)
- below the layout-selection-options on page 2 (warning-message, if there are slots being used in a layout but missing in the configuration of available layout slots)
- below the button "Edit result types" on page 2 (warning-message, if there are slots being used in result types but missing in the configuration of available layout slots)

If you also find this useful, I would like to provide this feature in an upcoming PR.

Kind regards, Joe

**Comment by kasperbolarsen** [comment]
Could you try to compare your idea to this , https://github.com/microsoft-search/pnp-modern-search/issues/3667 and see if there are any gaps ?

**Reply by jblueml** [reply]
Yes, I already checked [#3667](https://github.com/microsoft-search/pnp-modern-search/issues/3667). This would also be a useful extension - but a little more complex to implement.

My idea in a first step was just to parse the current layout- and results-types content and display warning-messages, if there are slots referred, that have not been configured.

_Example: Slots "Approver1", "Approver2" and "ApprovalDate" have been referred in my layout and result types_

On page 1:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/24583623/22b2c780-d116-4489-97dc-47374ee92e78)

On page 2:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/24583623/b3c4a3a9-78e5-4e1b-9c94-06252be36335)
